pkg/predicate/predicate.go

```diff
@@ -199,10 +199,23 @@ func IsDeleting() predicate.Predicate {
 }
 
 // AddTypePredicate returns a new slice which contains a type predicate and the given `predicates`.
-func AddTypePredicate(extensionType string, predicates []predicate.Predicate) []predicate.Predicate {
-	preds := make([]predicate.Predicate, 0, len(predicates)+1)
-	preds = append(preds, HasType(extensionType))
-	return append(preds, predicates...)
+// if more than one extensionTypes is given all given types are or combined
+func AddTypePredicate(predicates []predicate.Predicate, extensionTypes ...string) []predicate.Predicate {
```

Comment:
Looks great! But I think you have to adapt all the controllers in `pkg/controller` now, or?

Reply:
Damn, right, one more second ...